packages/react-native-windows-init/src/Cli.ts

```diff
@@ -274,36 +277,53 @@ async function getLatestMatchingRNWVersion(
   }
 }
 
-function installReactNativeWindows(version: string, useDevMode: boolean) {
-  console.log(
-    `Installing ${chalk.green('react-native-windows')}@${chalk.cyan(
-      version,
-    )}...`,
-  );
-
+function installReactNativeWindows(
+  version: string | undefined,
+  useDevMode: boolean,
+) {
   const cwd = process.cwd();
   const execOptions = argv.verbose ? {stdio: 'inherit' as 'inherit'} : {};
 
   if (useDevMode) {
-    execSync(
-      isProjectUsingYarn(cwd)
-        ? 'yarn link react-native-windows'
-        : 'npm link react-native-windows',
-      execOptions,
+    const packageCmd = isProjectUsingYarn(cwd) ? 'yarn' : 'npm';
+    execSync(`${packageCmd} link react-native-windows`, execOptions);
+    const rnwPkgJsonPath = path.resolve(
+      './node_modules/react-native-windows/package.json',
     );
-  } else {
-    const pkgJsonPath = findUp.sync('package.json', {cwd});
-    if (!pkgJsonPath) {
-      throw new Error('Unable to find package.json');
+    const rnwVersion = require(rnwPkgJsonPath).version;
+    if (version && version !== rnwVersion) {
+      console.error(
+        chalk.redBright(
+          `[Error] Requested react-native-windows version: '${version}' does not match version '${rnwVersion}' of the linked module. When using '--useDevMode' you do not need to pass a version. If you do, you should pass '--version ${rnwVersion}'`,
+        ),
```

Comment:
use `newError` instead to match other usages

Reply:
@asklar / @NickGerleman : My understanding of the code was that there were 2 kinds of errors in this code:
1) User errors (they have a specific exit code) print nice the error message to the console and terminate cleanly.
2) Assert/Bug style errors (they throw exception) which prints debug information like call stack etc...

Reply:
Note that this error is just here to improve the user experience for people using --useDevMode. Because otherwise yarn install will fail with a cryptic version mismatch error.
Note that I'll of course will update the wiki....

Reply:
I think Alex was referring to a function called "newError" in the CLI instead of throwing "new Error". That was a wrapper around "Ora" to show a failed icon next to a message. The functionality is internal to a separate package though.

Reply:
Ah,unfortunately that one is not in scope. it is part of package `@react-native-windows\cli`. This file is in `react-native-windows-init`. I dont' think it is worth adding a dependency here, or even factoring. I'll add a helper for all errors in this file since this is a common pattern in this file.

Reply:
ah fair I got confused with the other part of the cli